application/src/test/java/org/togetherjava/tjbot/commands/TopHelpersTest.java

```diff
@@ -0,0 +1,71 @@
+package org.togetherjava.tjbot.commands;
+
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.togetherjava.tjbot.db.Database;
+
+import java.nio.file.Path;
+import java.sql.SQLException;
+
+import static org.togetherjava.tjbot.db.generated.tables.MessageMetadata.MESSAGE_METADATA;
+
+final class TopHelpersTest {
+    private final long GUILD_ID = 2222_2222L;
+    private final long CHANNEL_ID = 3333_3333L;
+    private final String COUNT_OPTION = "count";
+
+    private Database database;
+    private SlashCommandEvent slashCommandEvent;
+    private TopHelpers topHelpers;
+
+    @BeforeEach
+    void setup(@TempDir Path tempDir) throws SQLException {
+        database = new Database("jdbc:sqlite:" + tempDir.toAbsolutePath() + "test.db");
+        topHelpers = new TopHelpers(database);
+
+        database.write(dsl -> {
+            dsl.insertInto(MESSAGE_METADATA, MESSAGE_METADATA.MESSAGE_ID, MESSAGE_METADATA.GUILD_ID,
+                    MESSAGE_METADATA.CHANNEL_ID, MESSAGE_METADATA.USER_ID,
+                    MESSAGE_METADATA.CREATE_TIMESTAMP)
+                // UserId 1L has posted 3 messages
+                .values(1L, GUILD_ID, CHANNEL_ID, 1L, System.currentTimeMillis())
+                .values(2L, GUILD_ID, CHANNEL_ID, 1L, System.currentTimeMillis())
+                .values(3L, GUILD_ID, CHANNEL_ID, 1L, System.currentTimeMillis())
+                // UserId 2L has posted 1 message
+                .values(4L, GUILD_ID, CHANNEL_ID, 2L, System.currentTimeMillis())
+                // UserId 3L has posted 2 messages
+                .values(5L, GUILD_ID, CHANNEL_ID, 3L, System.currentTimeMillis())
+                .values(6L, GUILD_ID, CHANNEL_ID, 3L, System.currentTimeMillis())
+                .execute();
+        });
+
+        slashCommandEvent = Mockito.mock(SlashCommandEvent.class, Mockito.RETURNS_DEEP_STUBS);
+        Mockito.when(slashCommandEvent.getGuild().getIdLong()).thenReturn(GUILD_ID);
+        Mockito.when(slashCommandEvent.getGuild().getMemberById(1L).getEffectiveName())
+            .thenReturn("ALPHA");
+        Mockito.when(slashCommandEvent.getGuild().getMemberById(2L).getEffectiveName())
+            .thenReturn("BETA");
+        Mockito.when(slashCommandEvent.getGuild().getMemberById(3L).getEffectiveName())
+            .thenReturn("GAMMA");
+    }
+
+    @Test
+    void get_top_helpers() {
```

Comment:
don't use snake_case here, Java methods should always be camelCase

Reply:
Can be done but JUnits are places where snake case naming has been accepted to aid readability. My bad I should have added a DisplayNameGenerator which would have highlighted the benefits.
If you have good naming, its better than keeping docs or comments in sync and is the very reason you would many a times see things like `fun 'some descriptive test name'()` in languages like Kotlin specifically used in test cases

References:
https://softwareengineering.stackexchange.com/questions/237561/naming-test-methods-in-java
https://github.com/junit-team/junit5/pull/1588 (JUnit 5 SPI for display name generation)
https://github.com/junit-team/junit5/issues/1595 (JUnit 5, failed attempt to create a camelCase based implementation as thre are too many edge cases)